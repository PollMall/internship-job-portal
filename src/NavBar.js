import React, { useContext } from 'react';
import {
  AppBar, Toolbar, Button, Typography, makeStyles, IconButton, Tooltip,
} from '@material-ui/core';
import HomeIcon from '@material-ui/icons/Home';
import { userAction } from './useUser';
import NavBarLink from './NavBarLink';
import { UserContext } from './UserProvider';

const useStyles = makeStyles(() => ({
  title: {
    flexGrow: 1,
  },

  menu: {
    textTransform: 'capitalize',
    maxWidth: 300,
  },
}));

function NavBar() {
  const classes = useStyles();
  const { state, dispatch } = useContext(UserContext);
  const handleLogout = React.useCallback(() => {
    dispatch({ type: userAction.LOGOUT });
  });

  return (
    <AppBar position="sticky">
      <Toolbar>
        <NavBarLink to="/home">
          <Tooltip title="Home">
            <IconButton color="inherit" edge="start">
              <HomeIcon />
            </IconButton>
          </Tooltip>
        </NavBarLink>
        <Typography data-testid="navbar--header-message" variant="h6" className={classes.title}>
          Hello
          {state.user && `, ${state.user.username}`}
        </Typography>
        {state.user && (
        <>
          {state.user.userRole?.name === 'user' && (
          <NavBarLink to="/profile">
            <Button data-testid="navbar--profile-btn" color="inherit">my profile</Button>
          </NavBarLink>
          )}
          <Button data-testid="navbar--logout-btn" color="inherit" onClick={handleLogout}>logout</Button>
        </>
        )}
      </Toolbar>
    </AppBar>
  );
}

export default NavBar;
