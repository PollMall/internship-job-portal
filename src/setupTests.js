// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { render } from '@testing-library/react';
import { mockApolloData } from './queries/MockApolloData';

Object.defineProperty(window, 'matchMedia', {
  value: () => ({
    matches: false,
    addListener: () => {},
    removeListener: () => {},
  }),
});

Object.defineProperty(window, 'alert', {
  value: jest.fn()
})

function makeMockApolloCalls(props) {
  return props.map((p) => {
    const { query, variables, data} = p;
    console.log(data);
    return {
      request: {
        query,
        variables: variables || {}
      },
      result: {
        data: data || {}
      }
    }
  })
}

const mockApolloCalls = makeMockApolloCalls(mockApolloData);

const AllProviders = ({ children }) => {
  return (
    <MockedProvider mocks={mockApolloCalls} addTypename={false}>
       <MemoryRouter>
         {children}
       </MemoryRouter>
     </MockedProvider>
  )
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllProviders, ...options })


export { customRender as render };