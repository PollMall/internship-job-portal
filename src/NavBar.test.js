import React from 'react';
import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { render } from './setupTests';
import NavBar from './NavBar';

describe('NavBar tests', () => {
  const user = {
    id: 1,
    username: 'user',
    password: 'user',
    userRole: {
      id: 1,
      name: 'user',
    },
  };
  it('should be rendered', () => {
    const { container } = render(<NavBar />);
    expect(container).toMatchSnapshot();
  });

  it('should have username displayed ', () => {
    render(<NavBar />, undefined, { user });
    const logoutBtn = screen.getByTestId('navbar--logout-btn');
    userEvent.click(logoutBtn);
    expect(screen.getByTestId('navbar--header-message')).toHaveTextContent(/^Hello$/);
  });

  it('should have link to profile page displayed ', () => {
    render(<NavBar />, undefined, { user });
    screen.getByTestId('navbar--profile-btn');
  });
});
