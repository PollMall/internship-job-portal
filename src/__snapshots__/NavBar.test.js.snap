// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar tests should be rendered 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <a
        class="makeStyles-link-4"
        href="/home"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          title="Home"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
      <h6
        class="MuiTypography-root makeStyles-title-1 MuiTypography-h6"
        data-testid="navbar--header-message"
      >
        Hello
      </h6>
    </div>
  </header>
</div>
`;
